views/hoc/AuthenticatedRoute.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useContext, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { AuthContext } from '../contexts/AuthContext'
+import { UserModel } from '../models'
+
+const AuthenticatedRoute = ({ children }) => {
+  const navigate = useNavigate()
+  const [user, setUser] = useContext(AuthContext)
+  const userId = window.sessionStorage.getItem('userId')
+
+  const fetchUser = async () => {
+    const res = await UserModel.findOne(userId)
+    if (res.status === 200) setUser(res.data)
+    else navigate('/login')
```

Comment:
we should have a `try/catch` here to navigate to the login route if the `findOne` call throws an error.